src/Sentry/SpanId.cs

```diff
@@ -7,58 +8,60 @@ namespace Sentry;
-    public static readonly SpanId Empty = new(EmptyValue);
+    public static readonly SpanId Empty = new(0);
 
     /// <summary>
     /// Creates a new instance of a Sentry span Id.
     /// </summary>
-    public SpanId(string value) => _value = value;
-
-    // This method is used to return a string with all zeroes in case
-    // the `_value` is equal to null.
-    // It can be equal to null because this is a struct and always has
-    // a parameterless constructor that evaluates to an instance with
-    // all fields initialized to default values.
-    // Effectively, using this method instead of just referencing `_value`
-    // makes the behavior more consistent, for example:
-    // default(SpanId).ToString() -> "0000000000000000"
-    // default(SpanId) == SpanId.Empty -> true
-    private string GetNormalizedValue() => !string.IsNullOrWhiteSpace(_value)
-        ? _value
-        : EmptyValue;
+    // TODO: mark as internal in version 4
+    public SpanId(string value) => long.TryParse(value, NumberStyles.HexNumber, null, out _value);
+
+    /// <summary>
+    /// Creates a new instance of a Sentry span Id.
+    /// </summary>
+    /// <param name="value"></param>
+    public SpanId(long value) => _value = value;
 
     /// <inheritdoc />
-    public bool Equals(SpanId other) => StringComparer.Ordinal.Equals(
-        GetNormalizedValue(),
-        other.GetNormalizedValue());
+    public bool Equals(SpanId other) => GetValue() == other.GetValue();
 
     /// <inheritdoc />
     public override bool Equals(object? obj) => obj is SpanId other && Equals(other);
 
     /// <inheritdoc />
-    public override int GetHashCode() => StringComparer.Ordinal.GetHashCode(GetNormalizedValue());
+    public override int GetHashCode() => StringComparer.Ordinal.GetHashCode(_value);
 
     /// <inheritdoc />
-    public override string ToString() => GetNormalizedValue();
+    public override string ToString() => _value.ToString("x8");
 
-    // Note: spans are sentry IDs with only 16 characters, rest being truncated.
-    // This is obviously a bad idea as it invalidates GUID's uniqueness properties
-    // (https://devblogs.microsoft.com/oldnewthing/20080627-00/?p=21823)
-    // but all other SDKs do it this way, so we have no choice but to comply.
     /// <summary>
     /// Generates a new Sentry ID.
     /// </summary>
-    public static SpanId Create() => new(Guid.NewGuid().ToString("n")[..16]);
+    public static SpanId Create()
+    {
+        var buf = new byte[8];
+        long random;
+
+        do
+        {
+            Random.NextBytes(buf);
+            random = BitConverter.ToInt64(buf, 0);
+        } while (random == 0);
+
+        return new SpanId(random);
+    }
 
     /// <inheritdoc />
-    public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? _) => writer.WriteStringValue(GetNormalizedValue());
+    public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? _) => writer.WriteStringValue(ToString());
```

Comment:
here we're still allocating the string before writing, is there a better way? Like writing each of the 8 bytes directly to the json writer instead?